lib/autoproj/daemon/package_repository.rb

```diff
@@ -39,6 +47,21 @@ def buildconf?
             def branch
                 vcs[:remote_branch] || vcs[:branch] || 'master'
             end
+
+            # @return [String]
+            def head_sha
+                pkg = autobuild
+                pkg.importdir ||= local_dir # TODO: What if importdir != srcdir?
+                pkg.importer.current_remote_commit(pkg, only_local: true)
+            end
+
+            # @return [Autoproj::FakePackage]
+            def autobuild
+                return Autobuild::Package[package] if Autobuild::Package[package]
```

Comment:
I'd like to get rid of `Autobuild::Package[]` at some point. Could you instead store the package in `self` ?

Reply:
I could but that's not the point here.. I have to use an existing instance (if an update was made) otherwise we get an error because packages are global. What I can do is use the `find_autobuild_package` rather than `Autobuild::Package`, would that suffice?

Reply:
> would that suffice?

Yep